wikimedia/wikipedia-ios
# Apple Pay - Donate form API calls

**Phabricator:** 
https://phabricator.wikimedia.org/T345847

### Notes
This PR adds the ability to do a few API calls for the donate form:

1. Fetching the [donate config file](https://donate.wikimedia.org/wiki/MediaWiki:AppsDonationConfig.json?action=raw).
2. Calling FR Tech's [payment networks](https://phabricator.wikimedia.org/T344860) endpoint. (Still in development)
3. Calling FRTech's [submit payment](https://phabricator.wikimedia.org/T344860) endpoint. (Still in development)

Since these are pretty basic calls*, I thought this was a good chance to spin up a new service in `WKData` and avoid the loop back to the app for networking. This new service is called `WKBasicService`.

_*FR Tech endpoints may eventually need an API key, but I don't have those details yet._

I am also fetching the config and payment methods endpoints at the same time as our current announcements endpoint fetch. I may modify it slightly in the future to trigger once our future [campaign config call](https://phabricator.wikimedia.org/T344024) goes through, to reduce unnecessary calls.

Note: https://github.com/wikimedia/wikipedia-ios/pull/4619/commits/b3aea1bdc197f820c6adf8e7bf5f24ece9467053 will come in handy while the FR Tech endpoints are still in development. So no need to revert before merging, but the goal is to eventually revert this commit.

### Test Steps

1. In the app and trigger an announcements fetch (generally just a background/foreground, then pull to refresh is needed on Explore). Confirm config values are printed out.
5. Confirm unit tests pass.
